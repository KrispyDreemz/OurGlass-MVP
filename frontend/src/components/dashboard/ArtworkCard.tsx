export interface Artwork {
  id: string;
  title: string;
  status: string;
  finalPrice?: number;
}

export default function ArtworkCard({ artwork }: { artwork: Artwork }) {
  return (
    <div className="border p-4 rounded bg-gray-800 text-white">
      <h3 className="font-semibold">{artwork.title}</h3>
      <p>Status: {artwork.status}</p>
      {artwork.finalPrice && (
        <p>Sold for ${artwork.finalPrice}</p>
      )}
    </div>
  );
}
