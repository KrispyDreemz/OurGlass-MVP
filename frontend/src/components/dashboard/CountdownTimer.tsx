'use client';
import { useEffect, useState } from 'react';

export default function CountdownTimer({ end }: { end: number }) {
  const [timeLeft, setTimeLeft] = useState(end - Date.now());

  useEffect(() => {
    const id = setInterval(() => setTimeLeft(end - Date.now()), 1000);
    return () => clearInterval(id);
  }, [end]);

  if (timeLeft <= 0) return <span className="text-red-400">Ended</span>;
  const hours = Math.floor(timeLeft / 3600000);
  const minutes = Math.floor((timeLeft % 3600000) / 60000);
  const seconds = Math.floor((timeLeft % 60000) / 1000);
  return (
    <span>{hours}h {minutes}m {seconds}s</span>
  );
}
