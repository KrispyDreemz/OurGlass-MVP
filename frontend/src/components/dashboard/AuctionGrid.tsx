'use client';
import { useState } from 'react';
import CountdownTimer from './CountdownTimer';
import BidModal from './BidModal';

export interface AuctionItem {
  id: string;
  title: string;
  image: string;
  currentBid: number;
  endsAt: number;
}

export default function AuctionGrid({ items }: { items: AuctionItem[] }) {
  const [selected, setSelected] = useState<AuctionItem | null>(null);

  return (
    <div>
      <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-4">
        {items.map((item) => (
          <div key={item.id} className="border rounded p-4 bg-gray-800 text-white">
            <img src={item.image} alt={item.title} className="w-full h-40 object-cover mb-2" />
            <h3 className="font-semibold">{item.title}</h3>
            <p>Bid: ${item.currentBid}</p>
            <CountdownTimer end={item.endsAt} />
            <button className="mt-2 px-2 py-1 bg-blue-600" onClick={() => setSelected(item)}>Bid Now</button>
          </div>
        ))}
      </div>
      <BidModal open={!!selected} onClose={() => setSelected(null)} onBid={() => {}} />
    </div>
  );
}
