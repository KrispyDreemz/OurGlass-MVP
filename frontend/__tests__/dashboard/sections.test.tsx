import { render } from '@testing-library/react';
import ArtworkCard from '@/components/dashboard/ArtworkCard';
import AuctionGrid from '@/components/dashboard/AuctionGrid';
import CountdownTimer from '@/components/dashboard/CountdownTimer';

const artwork = { id: 'a', title: 'Test', status: 'pending' };
const auctionItem = { id: 'x', title: 'Auction', image: '/img', currentBid: 10, endsAt: Date.now() + 1000 };
const bid = { id: 'b', title: 'Bid', amount: 10, status: 'active', endsAt: Date.now() + 1000 };

test('ArtworkCard snapshot', () => {
  const { container } = render(<ArtworkCard artwork={artwork} />);
  expect(container).toMatchSnapshot();
});

test('AuctionGrid snapshot', () => {
  const { container } = render(<AuctionGrid items={[auctionItem]} />);
  expect(container).toMatchSnapshot();
});

test('CountdownTimer snapshot', () => {
  const { container } = render(<CountdownTimer end={bid.endsAt} />);
  expect(container).toMatchSnapshot();
});
