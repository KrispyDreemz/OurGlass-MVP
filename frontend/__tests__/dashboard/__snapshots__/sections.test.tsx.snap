// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ArtworkCard snapshot 1`] = `
<div>
  <div
    class="border p-4 rounded bg-gray-800 text-white"
  >
    <h3
      class="font-semibold"
    >
      Test
    </h3>
    <p>
      Status: 
      pending
    </p>
  </div>
</div>
`;

exports[`AuctionGrid snapshot 1`] = `
<div>
  <div>
    <div
      class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-4"
    >
      <div
        class="border rounded p-4 bg-gray-800 text-white"
      >
        <img
          alt="Auction"
          class="w-full h-40 object-cover mb-2"
          src="/img"
        />
        <h3
          class="font-semibold"
        >
          Auction
        </h3>
        <p>
          Bid: $
          10
        </p>
        <span>
          0
          h 
          0
          m 
          0
          s
        </span>
        <button
          class="mt-2 px-2 py-1 bg-blue-600"
        >
          Bid Now
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CountdownTimer snapshot 1`] = `
<div>
  <span>
    0
    h 
    0
    m 
    0
    s
  </span>
</div>
`;
